examples/Basics/user_callable.py

```diff
@@ -0,0 +1,176 @@
+## Example for a user defined callable
+
+# This example is an extenssion of the recommenders example.
+# It implements an End-to-End example, that showes how to use the user defined callable in pure recommenders.
+
+# This examples assumes some basic familiarity with using BayBE.
+# We refer to [`campaign`](./campaign.md) for a more general and basic example.
+
+### Necessary imports for this example
+
+import pickle
+from typing import Optional
+
+from pandas import DataFrame
+
+from baybe import Campaign
+from baybe.objectives import SingleTargetObjective
+from baybe.parameters import NumericalDiscreteParameter, SubstanceParameter
+from baybe.recommenders import (
+    RandomRecommender,
+    SequentialGreedyRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace
+from baybe.surrogates import GaussianProcessSurrogate
+from baybe.targets import NumericalTarget
+from baybe.utils.dataframe import add_fake_results
+
+# Per default the initial recommender chosen is a random recommender.
+INITIAL_RECOMMENDER = RandomRecommender()
+
+### Surrogate models
+
+# This model uses available data to model the objective function as well as the uncertainty.
+# The surrogate model is then used by the acquisition function to make recommendations.
+
+# Per default a Gaussian Process is used
+SURROGATE_MODEL = GaussianProcessSurrogate()
+
+### Acquisition function
+
+# This function looks for points where measurements of the target value could improve the model.
+
+# Note that the qvailability of the acquisition functions might depend on the `batch_size`:
+#   - If `batch_size` is set to 1, all available acquisition functions can be chosen
+#   - If a larger value is chosen, only those that allow batching.
+#       That is, 'q'-variants of the acquisition functions must be chosen.
+
+# The default acquisition function is q-Expected Improvement.
+
+ACQ_FUNCTION = "qEI"
+
+
+### Callable function
+def callable_test(
+    recommender: Optional[SequentialGreedyRecommender] = None,
+    searchspace: Optional[SearchSpace] = None,
+    batch_size: Optional[int] = None,
+    train_x: Optional[DataFrame] = None,
+    train_y: Optional[DataFrame] = None,
+):
+    """Test the functionality of the implemented callable attribute.
+
+    Args:
+        recommender: Recommender to be used.
+        searchspace: The search space from which to recommend the points.
+        batch_size: The number of points to be recommended.
+        train_x: Optional training inputs for training a model.
+        train_y: Optional training labels for training a model.
+
+    This function prints all giving parameters and pickles the recommender object
```

Comment:
This part of the docstring needs to be between the heading and the "Args" section. Also, I'm wondering if it makes more sense to have this description not as a part of the function docstring but rather the general description.

Reply:
You mean it is better to make it part of the pre function comments, right?